File: python/python3_12/JavaScript/PythonParserBase.js
import antlr4 from "antlr4";

export default class PythonParserBase extends antlr4.Parser {
    constructor(input) {
        super(input);
    }

    isEqualToCurrentTokenText(tokenText) {
        return this.getCurrentToken().text === tokenText;
    }

    isnotEqualToCurrentTokenText(tokenText) {
        return !this.isEqualToCurrentTokenText(tokenText); // for compatibility with the Python 'not' logical operator
    }
}

File: python/python3_12/Python3/PythonParserBase.py
from antlr4 import InputStream, Parser
from typing import TextIO
import sys

class PythonParserBase(Parser):
    def __init__(self, input: InputStream, output: TextIO = sys.stdout):
        super().__init__(input, output)

    def isEqualToCurrentTokenText(self, tokenText: str) -> bool:
        return self.getCurrentToken().text == tokenText

    def isnotEqualToCurrentTokenText(self, tokenText: str) -> bool:
        return not self.isEqualToCurrentTokenText(tokenText) # for compatibility with the '!' logical operator in other languages
